src/server/hook_func.c

```diff
@@ -5321,6 +5328,38 @@ mk_deferred_hook_info(int index, int event, long long int tid)
 	return info;
 }
 
+/**
+ * @brief
+ *		check if there is any new pending action
+ *		if yes remove the current one
+ *	
+ * @param[in] minfo - pointer to mom info
+ * @param[in] pact - pointer to current hook action
+ * @param[in] j - index of pact in minfo->mi_action[]
+ * @param[in] event - action event to consider
+ *
+ * @return int
+ * @retval 0  - not found
+ * @retval 1  - found
+ */
+int
+check_for_latest_action(mominfo_t *minfo, mom_hook_action_t *pact, int j, int event)
```

Comment:
I don't see the 'j' index being used at all in this function. Did you mean to skip checking the minfo entries that match the 'j' index, since that is actually 'pact'?

Reply:
I was using it for a different purpose initially. and forgot to remove it later. But thanks, your suggestion is a good optimization. So fixed it in https://github.com/openpbs/openpbs/pull/1992/commits/d54ce17afb4e9d6cd1a3da070504726722e60e5c

Reply:
Ok, thanks. I'm seeing it more clearly now.